private/bufpkg/bufimage/bufimagemodify/internal/internal.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// fileOptionPath is the path prefix used for FileOptions.
+// All file option locations are preceded by a location
+// with a path set to the fileOptionPath.
+// https://github.com/protocolbuffers/protobuf/blob/053966b4959bdd21e4a24e657bcb97cb9de9e8a4/src/google/protobuf/descriptor.proto#L80
+var fileOptionPath = []int32{8}
+
+func RemoveLocationsFromSourceCodeInfo(sourceCodeInfo *descriptorpb.SourceCodeInfo, paths map[string]struct{}) error {
```

Comment:
This is moved from https://github.com/bufbuild/buf/blob/main/private/bufpkg/bufimage/bufimagemodify/file_option_sweeper.go#LL70C50-L70C50. The other functions in this file are also moved without changes